File: Easy/0219-Contains Duplicate II/Linear Scan.rs
impl Solution {
    pub fn contains_nearby_duplicate(nums: Vec<i32>, k: i32) -> bool {
        let len = nums.len();
        for i in 0..len {
            let mut j = i + 1;
            while j <= i + k as usize && j < len {
                if nums[i] == nums[j] {
                    return true;
                }
                j += 1;
            }
        }
        false
    }
}

File: Problemset/0035-Search Insert Position/Linear Scan.py
class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
        if not nums or target <= nums[0]:
            return 0
        if target > nums[-1]:
            return len(nums)
        for i in range(1, len(nums)):
            if target == nums[i] or (target > nums[i - 1] and target < nums[i]):
                return i
